sdk/src/types/block/core/wrapper.rs

```diff
@@ -16,11 +16,136 @@ use crate::types::block::{
+    /// Protocol version of the network to which this block belongs.
+    pub(crate) protocol_version: u8,
+    /// The identifier of the network to which this block belongs.
+    pub(crate) network_id: u64,
+    /// The time at which the block was issued. It is a Unix timestamp in nanoseconds.
+    pub(crate) issuing_time: Option<u64>,
+    /// The identifier of the slot to which this block commits.
+    pub(crate) slot_commitment_id: SlotCommitmentId,
+    /// The slot index of the latest finalized slot.
+    pub(crate) latest_finalized_slot: SlotIndex,
+    /// The identifier of the account that issued this block.
+    pub(crate) issuer_id: IssuerId,
+    /// The inner block.
+    pub(crate) block: Block,
+}
+
+impl BlockWrapperBuilder {
+    pub fn new(
+        protocol_version: u8,
+        network_id: u64,
+        slot_commitment_id: SlotCommitmentId,
+        latest_finalized_slot: SlotIndex,
+        issuer_id: IssuerId,
+        block: Block,
+    ) -> Self {
+        Self {
+            protocol_version,
+            network_id,
+            slot_commitment_id,
+            latest_finalized_slot,
+            issuer_id,
+            block,
+            issuing_time: Default::default(),
+        }
+    }
+
+    /// Updates protocol version.
+    #[inline(always)]
+    pub fn with_protocol_version(mut self, protocol_version: u8) -> Self {
+        self.protocol_version = protocol_version;
+        self
+    }
+
+    /// Updates network id
+    #[inline(always)]
+    pub fn with_network_id(mut self, network_id: u64) -> Self {
+        self.network_id = network_id;
+        self
+    }
+
+    /// Adds issuing time.
+    #[inline(always)]
+    pub fn with_issuing_time(mut self, issuing_time: impl Into<Option<u64>>) -> Self {
+        self.issuing_time = issuing_time.into();
+        self
+    }
+
+    /// Updates the slot commitment id.
+    #[inline(always)]
+    pub fn with_slot_commitment_id(mut self, slot_commitment_id: impl Into<SlotCommitmentId>) -> Self {
+        self.slot_commitment_id = slot_commitment_id.into();
+        self
+    }
+
+    /// Updates the latest finalized slot.
+    #[inline(always)]
+    pub fn with_latest_finalized_slot(mut self, latest_finalized_slot: impl Into<SlotIndex>) -> Self {
+        self.latest_finalized_slot = latest_finalized_slot.into();
+        self
+    }
+
+    /// Updates the issuer id.
+    #[inline(always)]
+    pub fn with_issuer_id(mut self, issuer_id: impl Into<IssuerId>) -> Self {
+        self.issuer_id = issuer_id.into();
+        self
+    }
+
+    /// Updates the block.
+    #[inline(always)]
+    pub fn with_block(mut self, block: impl Into<Block>) -> Self {
+        self.block = block.into();
+        self
+    }
+
+    pub(crate) fn header_hash(&self) -> [u8; 32] {
+        let mut bytes = [0u8; BlockHeader::LENGTH];
+        BlockHeader {
+            protocol_version: self.protocol_version,
+            network_id: self.network_id,
+            // TODO: what do here
```

Comment:
Why not make it required in the new() method?

Reply:
felt like it was something we should allow to be auto generated, but I dunno

Reply:
I feel like we should take an already built BlockHeader the same way we take an already built Block. This moves the problem of coming up with a value to somewhere else because I don't think it should be here